spec.go

```diff
@@ -5,6 +5,16 @@ import "context"
 // Logger is a simple interface describing services that emit messages to
 // gather certain runtime information.
 type Logger interface {
+	// Debug takes a format string and parameters and writes them in debug level.
+	Debug(format string, params ...interface{})
+	// Info takes a format string and parameters and writes them in info level.
+	Info(format string, params ...interface{})
+	// Warning takes a format string and parameters and writes them in warning level.
+	Warning(format string, params ...interface{})
+	// Error takes an error, a format string and parameters and writes them in
+	// error level. The error stack trace is written as "stack" value log
+	// entry.
+	Error(err error, format string, params ...interface{})
```

Comment:
It just struck my mind that on error we obviously want to include the stack trace as well. :slightly_smiling_face: